CMakeLists.txt

```
@@ -157,9 +154,19 @@ add_subdirectory(src)
 if (ENABLE_TESTS)
   add_subdirectory(tests)
 endif ()
-if (ENABLE_BENCHMARKS)
+
+if (ENABLE_DEVELOPER_BENCHMARKS)
   add_subdirectory(benchmarks)
+  if ((NOT CMAKE_BUILD_TYPE) OR (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Release"))
+    message("-- Warning: Please set CMAKE_BUILD_TYPE=Release when benchmarking!")
```

I think that it might be useful to run with BUILD_TYPE is set to Debug when/if one is actually debugging a benchmark implementation.  I like the warning message though and suggest it could be changed to:
```suggestion
    message("-- Warning: CMAKE_BUILD_TYPE not set to Release, benchmark information will not be reliable for this build!")
```